### PR TITLE
Add private_profile flag to users

### LINKED ISSUE #1066: User Visibility Controls
Users should be able to set their profiles to private, such that other users won't be able to see their feed or stats pages.

### PR DESCRIPTION
**Pull request by edpaget**
Allows users to mark their classification counts (and possibly future
parts of their profile) as private

Closes #1066

### CODE REVIEW

File: db/migrate/20150727212724_add_private_profile_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddPrivateProfileToUsers < ActiveRecord::Migration
+  def change
+    add_column :users, :private_profile, :boolean, index: { where: "(private_profile = true)" }, default: false
```
Comment (edpaget) [inline db/migrate/20150727212724_add_private_profile_to_users.rb]: Not sure if this should be true or false by default.

Comment (camallen) [inline db/migrate/20150727212724_add_private_profile_to_users.rb]: me either but i'm leaning to false by default with the option to go private.

**Comment by camallen** [comment]
LGTM

**Comment by edpaget** [comment]
Thanks just want to get some input from @DarrenMcRoy @chrissnyder @aliburchard about whether this should be public or private by default.

**Comment by DarrenMcRoy** [comment]
I agree with @camallen and vote public by default. I'll almost always vote for "opt-out" of features like this; the point of the feature is that it's good for the community, so it should be the default; the point of having it opt-out is that not everyone will want it, so they can turn it off.

**Comment by edpaget** [comment]
Do we know is good for the community? We have in the past been strong resistent to making classificaitons counts public.

**Comment by edpaget** [comment]
For instance, when science teams want 'top classifier' lists.

**Comment by chrissnyder** [comment]
I agree with Ed in that I don't think the answer here is as clear as it first appears. This is really the first time the Zoo has had any sort of public profile (right?), so it's uncharted territory for us.

Pros for public profile:

- Easier to manage (no decisions about what is public or not... everything is just public).
- More things to share, discuss.

Pros for private profile:

- Gives users a bit of a sense of control over their Zooniverse experience.
- Caters to those who want to keep their work more private.
- Potentially avoid problems like Ed mentions above.

Looking at the above, I'm leaning towards saying private is actually a better default mode. If we are in doubt that something should be public or not, that feels like an indication that it should indeed be private.

**Comment by camallen** [comment]
@mrniaboc thoughts on private profiles by default?

**Comment by mrniaboc** [comment]
Time to reference some actual data! When asked what things would make users increase/decrease their participation in the Zooniverse, the only one they reacted strongly against was competition with other users. This makes me lean heavily towards private as default.

participation-survey

**Comment by aliburchard** [comment]
Im all about stars being private by default. I like that @mrniaboc had data to auggest that people would prefer it, because I don't have anything other than a general sense of betrayal and annoyance when I've discovered my information has been made public without realizing it.

**Comment by eatyourgreens** [comment]
@aliburchard Makayla has a 500 page PhD thesis full of user research into that sense of betrayal and annoyance.

+1 to keeping stats private by default. If we switch to public by default, we should test the idea with our volunteers first.

**Comment by aliburchard** [comment]
@eatyourgreens that is a *lot* of pages...

**Comment by edpaget** [comment]
So it sounds like this should be bring by default. I'll update to make sure that that's the case.

**Comment by vrooje** [comment]
I know I'm a little late to the party, but I agree with @mrniaboc. Classification counts private by default... though I'd actually prefer to make the ribbons public by default. The ribbons aren't really a competition, just a description.

**Comment by DarrenMcRoy** [comment]
I can admit when I've been overruled. :-)

**Comment by edpaget** [comment]
@vrooje I'm not sure its possible to have one be public and one be private. And I think that should also be an optional thing, just since it is an change in established behaviour.

**Comment by vrooje** [comment]
Hmm, okay. I guess I'm thinking ahead to when we might eventually let people put mini-ribbons beneath their avatars on Talk. I've talked to some people who would be okay with that because it's based on percentages, but who would never want their classification *numbers* published.

**Comment by edpaget** [comment]
@camallen I think this should be good to go pending travis.

**Comment by edpaget** [comment]
@camallen don't know what's up with the failures. They're from the priority selection?

**Comment by camallen** [comment]
is this rebased of master?

**Pull request merged**